## Descripció
Exercicis d'Hipotesis Testing.

In [32]:
import numpy as np
import pandas as pd

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

from scipy.stats import t

In [33]:
Lligues = pd.read_csv('C:\\RAUL\\NISSAN\\GENERAL\\USB\\FORMACION\\BootCamp - Data Scientist_Analist\\IT Academy Cibernarium - Data Science\\Data Science Itinerario\\5.- Introducción al Test de Hipótesis y al Machine Learning\\Entrega\\Lligues europees UEFA.csv',  sep=';', encoding ='1252')
Lligues.head()

Rk            Squad Country  LgRk  MP   W  D  L  GF  GA  ...  Pts  Pts/G  \
0   1  Manchester City     ENG     1  37  28  6  3  96  24  ...   90   2.43   
1   2        Liverpool     ENG     2  36  26  8  2  89  24  ...   86   2.39   
2   3      Real Madrid     ESP     1  37  26  7  4  80  31  ...   85   2.30   
3   4    Bayern Munich     GER     1  34  24  5  5  97  37  ...   77   2.26   
4   5        Paris S-G     FRA     1  37  25  8  4  85  36  ...   83   2.24   

     xG   xGA   xGD  xGD/90     Last 5 Attendance          Top Team Scorer  \
0  86.1  26.8  59.3    1.60  W W W W D      52739     Kevin De Bruyne - 15   
1  84.6  33.1  51.4    1.43  W W W D W      53367       Mohamed Salah - 22   
2  73.0  45.8  27.2    0.73  W W L W D      40624       Karim Benzema - 27   
3  88.1  37.1  51.0    1.50  W W L D D      33176  Robert Lewandowski - 35   
4  71.6  38.1  33.4    0.90  W D D D W      41188       Kylian Mbappé - 25   

         Goalkeeper  
0           Ederson  
1           Alisson  
2  Thibaut Courtois  
3      Manuel Neuer  
4      Keylor Navas  

[5 rows x 21 columns]

# Exercici 1
#### Descarrega el dataset adjunt de dades oficials de la UEFA i selecciona un atribut del conjunt de dades. Calcula el p-valor i digues si rebutja la hipòtesi nul·la agafant un alfa de 5%. 

### **1.-** Formular la Hipòtesi Nul.la i la Alterativa:
**Ho:** Volem confirmar si la mitja de GA a la lliga alemanya és igual a la mitja de totes les lligues-->> μ_Ger = μ_Eur  
**Ha:** La Mitja de GA de la lliga alemanya igual o més gran de totes les lligues -->> μ_Ger > μ_Eur

### 2.- Recollida de dades en base el test d'hipòtesi.

In [34]:
GA_lligues_Eur=Lligues[['GA']]
GA_lliga_Ger=Lligues[Lligues['Country']=='GER' ]['GA']

μ_Eur=GA_lligues_Eur.mean()
μ_Ger=GA_lliga_Ger.mean()
S_Ger=np.std(GA_lliga_Ger)
n_Ger=18 #Total de la mostra
print('μ_Eur:'+str(μ_Eur)+' μ_Ger:'+str(μ_Ger))

μ_Eur:GA    50.877551
dtype: float64 μ_Ger:53.0


### 3.- Realització del Test Estadístic

In [35]:
T_Score=(μ_Eur-μ_Ger)/(S_Ger/np.sqrt(n_Ger))
p_value=t.sf(abs(T_Score), df=n_Ger-1) #trobem el p_value
print('p_value:',p_value,'T_Scorea:',T_Score)
0.2266283049085413

p_value: [0.20801061] T_Scorea: GA   -0.833681
dtype: float64


0.2266283049085413

### 4.- Decissió sobre la Hipòtesi Nul.la/Alternativa

  Com qu'el p_value (0.2080) és més gran que el valor α: 0.05, acceptem la Hipòtesi Nul.la --> μ_Ger = μ_Eur 

# Exercici 2
#### Amb el mateix dataset selecciona dos altres atributs del conjunt de dades. Calcula els p-valors i digues si rebutgen la hipòtesi nul·la agafant un alfa de 5%.

### **1.-** Formular la Hipòtesi Nul.la i la Alterativa:
**Ho:** Volem confirmar si la mitja de GA es igual a la mitja de GD -->> μ_GA = μ_GD  
**Ha:** Les mitjes de GA i GD no són iguals  -->> μ_GA != μ_GD 

### 2.- Recollida de dades en base el test d'hipòtesi.

In [36]:
from scipy.stats import shapiro
GA = Lligues[['GA']]
GD = Lligues[['GD']]

In [37]:
#Shapiro-Wilk Normality Test
def Distri_Normal(data): 
    stat, p = shapiro(data)
    print('stat=%.3f, p=%.3f' % (stat, p))
    if p > 0.05:
        print('Probably Gaussian')
    else:
        print('Probably not Gaussian')

In [38]:
Distri_Normal(GA)

stat=0.983, p=0.243
Probably Gaussian


In [39]:
Distri_Normal(GD)

stat=0.986, p=0.396
Probably Gaussian


### 3.- Realització del Test Estadístic

In [40]:
# Example of the Student's t-test
from scipy.stats import ttest_ind
data1 = GA
data2 = GD
stat, p = ttest_ind(data1, data2)
print('stat=%.3f, p_value=%.3f' % (stat, p))
if p > 0.05:
	print('Probably the same distribution')
else:
	print('Probably different distributions')

stat=16.899, p_value=0.000
Probably different distributions


### 4.- Decissió sobre la Hipòtesi Nul.la/Alternativa

  Com qu'el p_value (0.000) és zero i per tant inferior a α: 0.05, rebutgem la Hipòtesi Nul.la i validem la Hipòtesi alternativa --> μ_GA != μ_GD 

# Exercici 3
#### Continua amb el conjunt de dades adjunt i selecciona tres atributs del conjunt de dades. Calcula el p-valor i digues si rebutja la hipòtesi nul·la agafant un alfa de 5%.

### **1.-** Formular la Hipòtesi Nul.la i la Alterativa:
**Ho:** Volem confirmar si la mitja de GA, GD i xGD són iguals -->> μ_GA = μ_GD  = μ_xGD  
**Ha:** Les mitjes de GA, GD i xGD no són iguals

### 2.- Recollida de dades en base el test d'hipòtesi.

In [41]:
from scipy.stats import shapiro
xGD = Lligues[['xGD']]

In [42]:
Distri_Normal(Lligues[['xGD']])

stat=0.979, p=0.128
Probably Gaussian


### 3.- Realització del Test Estadístic

In [44]:
# Example of the Analysis of Variance Test
from scipy.stats import f_oneway
data1 = GA
data2 = GD
data3 = xGD
stat, p = f_oneway(data1, data2, data3)
print('stat=%.3f, p_value=%.3f' % (stat, p))
if p > 0.05:
	print('Probably the same distribution')
else:
	print('Probably different distributions')

stat=196.980, p_value=0.000
Probably different distributions


### 4.- Decissió sobre la Hipòtesi Nul.la/Alternativa

  Com qu'el p_value (0.000) és zero i per tant inferior a α: 0.05, rebutgem la Hipòtesi Nul.la i validem la Hipòtesi alternativa que indica que una o més mitjes són diferents.